### PR TITLE
server: Add create_game and join_game and required data structures and tests.

### PR DESCRIPTION
**Pull request by dvirk**

### CODE REVIEW

**Comment by mick-warehime** [comment]
Can you make sure all the ds_store files are deleted / gitignored

File: server/model/rooms.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        f'Tried to add ({room_name}) but a room with that name already '
+        f'exists.')
+    if icons is None:
+        icons = random.sample(ICONS, 2)
+
+    _room_dict[room_name] = GameState(room_name, [first_player], 'Team 1', 0,
+                                      icons[0], [], 'Team 2', 0, icons[1])
+
+
+def get_room_data(room_name: str, field: GameFields) -> Any:
```
Comment (mick-warehime) [inline server/model/rooms.py:44]: Why not get_room() --> room 
And update_room(room_name, room)

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
[the remark this answers is not in the document]
Comment (dvirk) [reply .github/workflows/pytest.yml:19]: Why is that necessary?

File: app/app.py [hunk 3]
```diff
@@ -2,13 +2,19 @@
...
 
 from flask import Flask, render_template
 from flask_cors import CORS
 from flask_socketio import SocketIO
 from flask_socketio import emit
 
-from icons import ICONS
+from app.icons import ICONS
+from server.app_utils import validate_fields
+from server.model.fields import (ROOM_NAME, ERROR, PLAYER_NAME, GAME_STATE,
```
[the remark this answers is not in the document]
Comment (dvirk) [reply app/app.py:14]: I will import emit

File: server/model/rooms.py (hunk 1 above)
Comment (dvirk) [reply server/model/rooms.py:44]: I will pass namedtuples

**Pull request merged**